src/librustc/ty/sty.rs

```diff
@@ -2299,20 +2298,7 @@ impl<'tcx> Const<'tcx> {
         ty: Ty<'tcx>,
     ) -> Option<u128> {
         assert_eq!(self.ty, ty);
-        // This is purely an optimization -- layout_of is a pretty expensive operation,
-        // but if we can determine the size without calling it, we don't need all that complexity
-        // (hashing, caching, etc.). As such, try to skip it.
-        let size = match ty.kind {
-            ty::Bool => Size::from_bytes(1),
-            ty::Char => Size::from_bytes(4),
-            ty::Int(ity) => {
-                Integer::from_attr(&tcx, SignedInt(ity)).size()
-            }
-            ty::Uint(uty) => {
-                Integer::from_attr(&tcx, UnsignedInt(uty)).size()
-            }
-            _ => tcx.layout_of(param_env.with_reveal_all().and(ty)).ok()?.size,
-        };
```

Comment:
A quick grep shows a few more sites that call `layout_of` just tog et the size or alignment (and there's likely more but split across multiple lines). So it might still be worth having `size_and_align_of(Ty<'tcx>)` somewhere with an optimization like this -- maybe?

Reply:
yea, this could just be a method on `TyCtxt`